homeassistant/components/zwave_js/addon.py

```diff
@@ -0,0 +1,246 @@
+            LOGGER.warning("Z-Wave JS add-on is not installed. Installing add-on")
+            self._install_task = self._async_schedule_addon_operation(
+                self.async_install_addon,
+                partial(self.async_setup_addon, usb_path, network_key),
+            )
+        return self._install_task
+
+    @api_error("Failed to uninstall the Z-Wave JS add-on")
+    async def async_uninstall_addon(self) -> None:
+        """Uninstall the Z-Wave JS add-on."""
+        await async_uninstall_addon(self._hass, ADDON_SLUG)
+
+    @api_error("Failed to update the Z-Wave JS add-on")
+    async def async_update_addon(self) -> None:
+        """Update the Z-Wave JS add-on if needed."""
+        addon_info = await self.async_get_addon_info()
+        addon_version = addon_info["version"]
+        update_available = addon_info["update_available"]
+
+        if addon_version is None:
+            raise AddonError("Z-Wave JS add-on is not installed")
+
+        if not update_available:
+            return
+
+        await async_update_addon(self._hass, ADDON_SLUG)
+
+    @callback
+    def async_schedule_update_addon(self) -> asyncio.Task:
+        """Schedule a task that updates and sets up the Z-Wave JS add-on.
+
+        Only schedule a new update task if the there's no running task.
+        """
+        if not self._update_task or self._update_task.done():
+            LOGGER.warning("Trying to update the Z-Wave JS add-on")
+            self._update_task = self._async_schedule_addon_operation(
+                self.async_create_snapshot, self.async_update_addon
+            )
+        return self._update_task
+
+    @api_error("Failed to start the Z-Wave JS add-on")
+    async def async_start_addon(self) -> None:
+        """Start the Z-Wave JS add-on."""
+        await async_start_addon(self._hass, ADDON_SLUG)
+
+    @api_error("Failed to stop the Z-Wave JS add-on")
+    async def async_stop_addon(self) -> None:
+        """Stop the Z-Wave JS add-on."""
+        await async_stop_addon(self._hass, ADDON_SLUG)
+
+    async def async_setup_addon(self, usb_path: str, network_key: str) -> None:
+        """Configure and start Z-Wave JS add-on."""
+        addon_options = await self.async_get_addon_options()
+
+        new_addon_options = {
+            CONF_ADDON_DEVICE: usb_path,
+            CONF_ADDON_NETWORK_KEY: network_key,
+        }
+
+        if new_addon_options != addon_options:
+            await self.async_set_addon_options(new_addon_options)
+
+        await self.async_start_addon()
+
+    @callback
+    def async_schedule_setup_addon(
+        self, usb_path: str, network_key: str
+    ) -> asyncio.Task:
+        """Schedule a task that configures and starts the Z-Wave JS add-on.
+
+        Only schedule a new setup task if the there's no running task.
+        """
+        if not self._setup_task or self._setup_task.done():
+            LOGGER.warning("Z-Wave JS add-on is not running. Starting add-on")
```

Comment:
```suggestion
            LOGGER.debug("Z-Wave JS add-on is not running. Starting add-on")
```
Does not seem important enough to warn about